### PR TITLE
Datepicker improvements

### PR DESCRIPTION
**Pull request by nikgraf**
solves: https://github.com/nikgraf/belle/issues/255

### CODE REVIEW

File: src/__tests__/DatePicker-test.js [hunk 1]
```diff
@@ -283,7 +283,7 @@ describe('DatePicker', () => {
     const day = TestUtils.scryRenderedDOMComponentsWithClass(datePicker, 'day_test')[8];
     TestUtils.Simulate.focus(day);
     TestUtils.Simulate.mouseDown(day, { button: 0 });
-    expect(datePicker.state.selectedDate).toBe(compSelectedDate);
+    expect(datePicker.state.selectedDate.getTime()).toBe(compSelectedDate.getTime());
   });
```
Comment (jpuri) [inline src/__tests__/DatePicker-test.js]: Why prefer time over date ?

File: src/utils/date-helpers.js [hunk 2]
```diff
@@ -59,10 +59,42 @@ export const getLocaleData = (locale) => {
...
+ * @param year {number} - any year
+ * @param month {number} - can be between 1 and 12
+ * @param day {number} - can be between 1 and 31 depending on the month
+ * @returns {date} - the parse date
+ */
+export const getDateForDateKey = (dateKey) => {
+  const [year, month, day] = dateKey.split('-');
+  return new Date(parseInt(year, 10), parseInt(month, 10) - 1, parseInt(day, 10));
+};
+
```
Comment (jpuri) [inline src/utils/date-helpers.js]: Hey @nikgraf , why add function for something date constructor already does for you.

File: src/__tests__/DatePicker-test.js (hunk 1 above)
Comment (nikgraf) [inline src/__tests__/DatePicker-test.js]: because comparing dates actually does not work

File: src/utils/date-helpers.js (hunk 2 above)
Comment (nikgraf) [inline src/utils/date-helpers.js]: It does not work in Firefox without it

File: src/__tests__/DatePicker-test.js (hunk 1 above)
Comment (jpuri) [inline src/__tests__/DatePicker-test.js]: Good point :smile:

**Pull request merged**